### DISCUSSION TITLE
Node.js 18 is currently available in preview for GAE Flex.

### DISCUSSION THREAD

**Discussion by kennethye1** [GoogleCloudPlatform/nodejs-docker#263] [Announcements]
Requirements:

- Node.js 18 uses Ubuntu 22 base image
- You **must** specify the operating system in your `app.yaml` via
```
runtime_config:
   operating_system: ubuntu22
```
- Currently only supported with `gcloud beta app deploy` with latest gcloud versions i.e. `gcloud` version >= 417.0.1.
- **Optional**: You can specify the runtime version in `app.yaml` under `runtime_config` with property `runtime_version` in addition to the `package.json`. The `runtime_version` is optional but if specified will take precedence over `engines` version. 

We are working on releasing it for `gcloud app deploy`.

**Comment by vvvctr** [comment]
Hey @kennethye1, this has already been released, hasn't it?
I see a working example here: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/main/appengine/hello-world/flexible
